R/markdown-link.R

```diff
@@ -174,6 +185,100 @@ parse_link <- function(destination, contents, state) {
   }
 }
 
+resolve_link_package <- function(topic, me = NULL, pkgdir = NULL) {
+  me <- me %||% roxy_meta_get("current_package")
+  # this is  from the roxygen2 tests, should not happen on a real package
+  if (is.null(me) || is.na(me) || me == "") return(NA_character_)
+
+  # if it is in the current package, then no need for package name, right?
+  if (has_topic(topic, me)) return(NA_character_)
+
+  # try packages in depends, imports, suggests first, error on name clashes
+  pkgs <- local_pkg_deps(pkgdir)
+
+  pkg_has_topic <- pkgs[map_lgl(pkgs, has_topic, topic = topic)]
+  pkg_has_topic <- map_chr(pkg_has_topic, function(p) {
+    p0 <- find_reexport_source(topic, p)
+    if (is.na(p0)) p else p0
+  })
+  pkg_has_topic <- unique(pkg_has_topic)
+  base <- base_packages()
+  if (length(pkg_has_topic) == 0) {
+    # fall through to check base packages as well
+  } else if (length(pkg_has_topic) == 1) {
+    if (pkg_has_topic %in% base) {
+      return(NA_character_)
+    } else {
+      return(pkg_has_topic)
+    }
+  } else {
+    cli::cli_warn(c(
```

Comment:
Can you use `warn_roxy_topic()` here (and below)?

Reply:
I don't have the topic here I think. I am not even sure that we have a topic at this point. Maybe I could use `warn_roxy_tag()` if we pass the tag through the machinery?

Reply:
Oh I pattern matched on the wrong `topic`.  I think it would be worth passing all the necessary info down to so `warn_roxy_tag()` can give a file name/line number.

Reply:
This is was pretty easy in the end.